src/ServiceControl.Config/UI/Shell/VersionCheckerHelper.cs

```diff
@@ -0,0 +1,90 @@
+﻿namespace ServiceControl.Config.UI.Shell
+{
+    using Newtonsoft.Json;
+    using ServiceControlInstaller.Engine.Configuration.ServiceControl;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+
+    public static class VersionCheckerHelper
+    {
+        public static async Task<Release> GetDownloadUrlForNextVersionToUpdate(string currentVersion)
+        {
+            Version current = new Version(currentVersion);
+            List<Release> releases = await GetVersionInformation().ConfigureAwait(false);
+
+            Release topversion = releases.Select(t => (t.Version, t)).Max().t;
+            
+            UpgradeInfo upgradeInfo = UpgradeControl.GetUpgradeInfoForTargetVersion(topversion.Version, current);
```

Comment:
I'm not certain that this gives us want we want. Here's why:

The current major is v4. Whenever we release v5, it will show up on the web and this code will find it. A call to `UpgradeControl.GetUpgradeInfoForTargetVersion(5.x, 4.x)` will return a recommended version of 5.x. This is because we don't have minimum/recommended upgrade versions for version 5. We won't have those until we release version 5, by which time, it will be too late because you'll have to upgrade to version 5 to have it.

The `UpgradeControl` class is designed to prevent updating to a major if the current instance version is not compatible. It does not help if you're looking at the next download and trying to figure out if it will be compatible with the current version.

To make this work properly, we'd need to put the restrictions in the release info json on the web. Then we could list upgrade info there and have the system make a recommendation for a specific download using that.

Reply:
So what are you suggesting we do?

Reply:
@seanfarmar for this enhancement release, I think we just need to check if there is a version that is higher than the currently installed one. I would not use the `UpgradeControl` class. A comparison of `Version`s is good enough for now.